server/src/test/java/org/apache/cassandra/sidecar/lifecycle/LifecycleManagerTest.java

```diff
@@ -0,0 +1,300 @@
+        mockLifecycleProvider = mock(LifecycleProvider.class);
+        when(metadataFetcher.instance(TEST_HOST)).thenReturn(TEST_HOST_META);
+        when(metadataFetcher.instance(TEST_HOST_2)).thenReturn(TEST_HOST_2_META);
+    }
+
+    @AfterEach
+    void cleanup()
+    {
+        TestResourceReaper.create().with(vertx).with(executorPools).close();
+    }
+
+    @Test
+    void testGetLifecycleInfoWithNoTaskSubmitted()
+    {
+        LifecycleManager lifecycleManager = new LifecycleManager(metadataFetcher, mockLifecycleProvider, executorPools);
+        when(mockLifecycleProvider.isRunning(TEST_HOST_META)).thenReturn(false);
+
+        LifecycleInfoResponse response = lifecycleManager.getLifecycleInfo(TEST_HOST);
+        assertThat(response.currentState()).isEqualTo(LifecycleCassandraState.STOPPED);
+        assertThat(response.desiredState()).isEqualTo(LifecycleCassandraState.UNKNOWN);
+        assertThat(response.status()).isEqualTo(LifecycleStatus.UNDEFINED);
+        assertThat(response.lastUpdate()).isEqualTo("No lifecycle task submitted for this instance yet.");
+    }
+
+    @Test
+    void testSubmittedTaskSucceeds() throws LifecycleTaskConflictException, InterruptedException
+    {
+        // Submit slow start task
+        when(mockLifecycleProvider.isRunning(TEST_HOST_META)).thenReturn(false);
+        CountDownLatch startLatch = slowCassandraStart();
+        LifecycleManager lifecycleManager = new LifecycleManager(metadataFetcher, mockLifecycleProvider, executorPools);
+
+        LifecycleInfoResponse actualResponse = lifecycleManager.updateDesiredState(TEST_HOST, LifecycleCassandraState.RUNNING);
+        LifecycleInfoResponse expectedResponse = new LifecycleInfoResponse(LifecycleCassandraState.STOPPED, LifecycleCassandraState.RUNNING,
+                                                                           LifecycleStatus.CONVERGING,
+                                                                           "Submitting start task for instance");
+        assertThat(actualResponse).isEqualTo(expectedResponse);
+
+        // Wait for the task to complete
+        startLatch.countDown();
+        when(mockLifecycleProvider.isRunning(TEST_HOST_META)).thenReturn(true);
+        Thread.sleep(200);
+
+        // Check status was updated
+        actualResponse = lifecycleManager.getLifecycleInfo(TEST_HOST);
+        expectedResponse = new LifecycleInfoResponse(LifecycleCassandraState.RUNNING, LifecycleCassandraState.RUNNING,
+                                                     LifecycleStatus.CONVERGED,
+                                                     "Instance has started");
+        assertThat(actualResponse).isEqualTo(expectedResponse);
+
+        // Attempt to start the instance again, should be no-op since instance is already running
+        actualResponse = lifecycleManager.updateDesiredState(TEST_HOST, LifecycleCassandraState.RUNNING);
+        assertThat(actualResponse).isEqualTo(expectedResponse);
```

Comment:
We like to avoid doing thread sleep and we actually have a helper method that allows us to wait in a loop
```suggestion
        
        // Check status was updated
        LifecycleInfoResponse actualResponseAfterStart = lifecycleManager.getLifecycleInfo(TEST_HOST);
        LifecycleInfoResponse expectedResponseAfterStart = new LifecycleInfoResponse(LifecycleCassandraState.RUNNING, LifecycleCassandraState.RUNNING,
                                                     LifecycleStatus.CONVERGED,
                                                     "Instance has started");
        loopAssert(1, 10, () -> assertThat(actualResponseAfterStart).isEqualTo(expectedResponseAfterStart));

        // Attempt to start the instance again, should be no-op since instance is already running
        actualResponse = lifecycleManager.updateDesiredState(TEST_HOST, LifecycleCassandraState.RUNNING);
        assertThat(actualResponse).isEqualTo(expectedResponseAfterStart);
```